docs/guide.rst

```diff
@@ -86,3 +86,141 @@ options::
 
 Go on and explore. Don't be shy and ask, ``--help`` is eager to
 answer all your questions ;-)
+
+
+Under The Hood
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now let's have a brief look under the hood. Similarly as with
+``git``, there is a special ``.fmf`` directory which marks root of
+the fmf metadata tree. Use the ``init`` command to initialize it::
+
+    tmt init
+
+Do not forget to include this special ``.fmf`` directory in your
+commits, it is essential for building the fmf tree structure which
+is created from all ``*.fmf`` files discovered under the fmf root.
+
+
+Plans
+------------------------------------------------------------------
+
+As we've seen above, in order to enable a simple test the
+following plan is just enough::
+
+    execute:
+        script: foo --version
+
+Store these two lines in a ``*.fmf`` file and that's it. Name and
+location of the file is completely up to you, plans are recognized
+by the ``execute`` key which is required. Once the newly created
+plan is submitted to the CI system test script will be executed.
+
+:ref:`/spec/plans` are used to enable testing and group relevant
+tests together. They describe how to :ref:`/spec/plans/discover`
+tests for execution, how to :ref:`/spec/plans/provision` the
+environment, how to :ref:`/spec/plans/prepare` it for testing, how
+to :ref:`/spec/plans/execute` tests, :ref:`/spec/plans/report`
+results and finally how to :ref:`/spec/plans/finish` the test job.
+
+Here's an example of a slightly more complex plan which changes
+the default provision method to container to speed up the testing
+process and ensures that an additional package is installed before
+the testing starts::
+
+    provision:
+        how: container
```

Comment:
As Denis mentioned, I think it would be nice to have a few sentences mentioning that each step can be implemented through multiple methods and the desired one is chosen using `how` keyword. (just a bit more general explanation before we go to this example)

Reply:
Makes sense, added a new paragraph explaining the concept of multiple implementations.